src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -306,13 +314,67 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                         End If
                     Loop
 
+                    If (compareKind And TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) = 0 Then
+                        Return HasSameTypeArgumentCustomModifiers(DirectCast(t1, NamedTypeSymbol), DirectCast(t2, NamedTypeSymbol))
+                    End If
```

Comment:
It looks like for nested types we will miss this check on containers. It should be incorporated into the Do/Loop above.